pkg/scheduler/scheduler.go

```diff
@@ -608,6 +612,13 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 		return
 	}
 
+	// At the end of a successful scheduling cycle, pop and move up Pods if needed.
+	if len(podsToActivate.Map) != 0 {
```

Comment:
shouldn't we run this even if we were not successful in scheduling the pod, not only on success? the failed to schedule pod may want another pod to scheduler first and so it may want to activate it.

Reply:
It's debatable but my option is that "allowing the failed pod to activate pods" makes the mechanics complicated/chaotic and thus hard to use/debug, sometimes may generate a cycling dependency.

Take a group of pods as an example, some are "critical path" and some are not. I'd like the "critical" pod (e.g., a spark driver) to trigger the non-critical (executors) pods' activating; instead of the other way around. In terms of gang scheduling, although the implementation differs, an efficient way to do this is to let the Nth pod enters into waiting, while also activating other pods.

In other words, for a group of pods, I'd assume them to be inter-dependent in a unidirectional manner (instead of bidirectional).

Reply:
Ok, sounds a bit ad-hoc either way. In any case, lets clarify those semantics, I left a comment about that.